src/Symfony/Component/HttpFoundation/UriSigner.php

```diff
@@ -72,7 +92,15 @@ public function check(string $uri): bool
         $hash = $params[$this->parameter];
         unset($params[$this->parameter]);
 
-        return hash_equals($this->computeHash($this->buildUrl($url, $params)), $hash);
+        if (!hash_equals($this->computeHash($this->buildUrl($url, $params)), $hash)) {
+            return false;
+        }
+
+        if ($expiration = $params[$this->expirationParameter] ?? false) {
+            return !(time() >= (int) $expiration);
```

Comment:
the cast to int is dangerous on 32-bits system, better let php handle this for us:
```suggestion
            return time() < $expiration;
```